src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -312,36 +338,56 @@ private IConfidentialClientApplication BuildConfidentialClientApplication()
                 request.PathBase,
                 microsoftIdentityOptions.CallbackPath.Value ?? string.Empty);
 
-            if (!applicationOptions.Instance.EndsWith("/"))
-                applicationOptions.Instance += "/";
-
-            string authority ;
-            IConfidentialClientApplication app = null;
-
-            if (microsoftIdentityOptions.IsB2C)
+           if (string.IsNullOrEmpty(applicationOptions.Instance) &&
```

Comment:
I don't have the context around what `applicationOptions` and `microsoftIdentityOptions` are, but it looks to me that it violates the Single Responsability Principle. This class knows too much about options - this is a code smell that you are missing an object whose responsability is to know how to combine these options.

Reply:
agree. will do this in a separate PR

---
In reply to: [385592741](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r385592741) [](ancestors = 385592741)